ecl/hql/hqlutil.cpp

```diff
@@ -4794,9 +4794,10 @@ bool BitfieldPacker::checkSpaceAvailable(unsigned & thisBitOffset, unsigned & th
 {
     bool fitted = true;
     thisBits = type->getBitSize();
-    if (thisBits > bitsRemaining)
+    ITypeInfo * storeType = type->queryChildType();
+    if ((thisBits > bitsRemaining) || (activeType && storeType != activeType))
```

Comment:
@ghalliday Minor: Is it necessary to check activeType (before the storeType != activeType):  if activeType==null, bitsRemaining is also always 0, so, the (thisBits>bitsRemaining) will ensure second part IF is never executed.